[bug]: backtrace.c: Compilation failure due to use of removed type

### Severity

Major

### Versions

Git

### Components/Modules

backtrace.c

### Operating Environment

Arch Linux, gcc 16.1.1

### Frequency of Occurrence

Constant

### Issue Description

Compilation failure due to use of removed type

### Relevant log output

```shell
[CCi] test_websocket_client.i -> test_websocket_client.o
   [LD] test_websocket_client.o -> test_websocket_client.so
   [CPP] test_xml_escape.c -> test_xml_escape.i
   [CCi] test_xml_escape.i -> test_xml_escape.o
   [LD] test_xml_escape.o -> test_xml_escape.so
   [CCi] backtrace.i -> backtrace.o
backtrace.c: In function ‘process_section’:
backtrace.c:130:9: error: unknown type name ‘bfd_boolean’
  130 |         bfd_boolean line_found = 0;
      |         ^~~~~~~~~~~
make[1]: *** [/usr/src/asterisk-22.10.1/Makefile.rules:157: backtrace.o] Error 1
make: *** [Makefile:402: main] Error 2
```

### Asterisk Issue Guidelines

- [x] Yes, I have read the Asterisk Issue Guidelines

--- a/main/backtrace.c
+++ b/main/backtrace.c
@@ -127,7 +127,7 @@ static void process_section(bfd *bfdobj, asection *section, void *obj)
 	bfd_vma offset;
 	bfd_vma vma;
 	bfd_size_type size;
-	bfd_boolean line_found = 0;
+	bool line_found = 0;
 	const char *fn;
 	int inlined = 0;
 

